src/injector/inject.ts

```diff
@@ -344,9 +345,12 @@ export function inject (vm: ChibiCompatibleVM) {
             return originalGetOrderFunc.call(this, extensions, ...args);
         };
     }
-
+    // Expose a context for developers.
+    if (!window.chibi.settings.dontExposeCtx) {
+        window.Scratch = makeCtx(vm);
```

Comment:
Exposing Scratch objects in the first place is not a good idea. We should only consider exposing this object when loading non-sandboxed extensions to ensure compatibility.
Likewise, this code conflicts with the logic of restoring the original Scratch object.

Reply:
No it does not contain the functionality of `Scratch.extension.register`. Considering add an extra option about this? I will remove this line later tho.